Aztec/Classes/TextKit/TextStorage.swift

```diff
@@ -211,6 +212,47 @@ open class TextStorage: NSTextStorage {
         return finalString
     }
 
+    /// Preprocesses an attributed string that is missing a `headingRepresentation` attribute for insertion in the storage.
+    ///
+    /// - Important: This method adds the `headingRepresentation` attribute if it determines the string should contain it.
+    ///  This works around a problem where autocorrected text didn't contain the attribute. This may change in future versions.
+    ///
+    /// - Parameters:
+    ///     - attributedString: the string we need to preprocess.
+    ///
+    /// - Returns: the preprocessed string.
+    ///
+    fileprivate func preprocessHeadingsForInsertion(_ attributedString: NSAttributedString) -> NSAttributedString {
+        // Ref. https://github.com/wordpress-mobile/AztecEditor-iOS/pull/1334
+
+        guard textStore.length > 0, attributedString.length > 0 else {
+            return attributedString
+        }
```

Comment:
[not a blocker] Is this necessary? 
I wasn't sure what purpose this had, e.g. if it was here to make the function more efficient by aborting early, or because it prevented a crash on empty strings on later lines.
I couldn't find a scenario where it led to an issue, so I'm perfectly happy leaving it in. I was just curious of it's precise role here.

Reply:
I believe it's necessary and should prevent a crash when deleting characters:

1. Start a heading and type some characters
2. Erase a character

`attributedString.attribute(.headingRepresentation, at: 0, effectiveRange: nil)` will crash with an out-of-bounds exception because we tried to get an attribute at location `0` on a string that has zero characters.

The `textStore.length` check could prevent the same exception when it's empty as well. 👍